xarray/core/dataset.py

```diff
@@ -5401,6 +5401,13 @@ def transpose(
         numpy.transpose
         DataArray.transpose
         """
+        # Raise error if list is passed as dims
+        if (len(dims) > 0) and (isinstance(dims[0], list)):
+            list_fix = [f"'{x}', " if isinstance(x, str) else f"{x}, " for x in dims[0]]
+            raise TypeError(
+                f'transpose requires dims to be of hashable type. Expected "{"".join(list_fix)[:-2]}". Received "{dims[0]}" instead'
```

Comment:
```suggestion
                f'transpose requires dims to be passed as multiple arguments. Expected "{"".join(list_fix)[:-2]}". Received "{dims[0]}" instead'
```

(maybe a bit more approachable, `Hashable` may be confusing for newbies, though is more correct...)

Why the `[:-2]`?

Reply:
[:-2] was meant to get rid of the last comma and whitespace but I'm now realizing I can accomplish that by removing them from the list_fix and making the join statement ", ".join(list_fix).